src/dolphin/os/OSInterrupt.h

```diff
@@ -52,17 +63,56 @@ typedef enum {
 #define OS_INTRMASK_EXI_1_EXT (0x80000000U >> OS_INTR_EXI_1_EXT)
 #define OS_INTRMASK_EXI_2_EXI (0x80000000U >> OS_INTR_EXI_2_EXI)
 #define OS_INTRMASK_EXI_2_TC (0x80000000U >> OS_INTR_EXI_2_TC)
+#define OS_INTRMASK_PI_CP (0x80000000U >> OS_INTR_PI_CP)
+#define OS_INTRMASK_PI_PE_TOKEN (0x80000000U >> OS_INTR_PI_PE_TOKEN)
+#define OS_INTRMASK_PI_PE_FINISH (0x80000000U >> OS_INTR_PI_PE_FINISH)
+#define OS_INTRMASK_PI_SI (0x80000000U >> OS_INTR_PI_SI)
+#define OS_INTRMASK_PI_DI (0x80000000U >> OS_INTR_PI_DI)
+#define OS_INTRMASK_PI_RSW (0x80000000U >> OS_INTR_PI_RSW)
+#define OS_INTRMASK_PI_ERROR (0x80000000U >> OS_INTR_PI_ERROR)
+#define OS_INTRMASK_PI_VI (0x80000000U >> OS_INTR_PI_VI)
 #define OS_INTRMASK_PI_DEBUG (0x80000000U >> OS_INTR_PI_DEBUG)
+#define OS_INTRMASK_PI_HSP (0x80000000U >> OS_INTR_PI_HSP)
+
+#define OS_INTRMASK_MEM                                                        \
+    (OS_INTRMASK_MEM_0 | OS_INTRMASK_MEM_1 | OS_INTRMASK_MEM_2 |               \
+     OS_INTRMASK_MEM_3 | OS_INTRMASK_MEM_ADDRESS)
+
+#define OS_INTRMASK_AI (OS_INTRMASK_AI_AI)
+
+#define OS_INTRMASK_DSP                                                        \
+    (OS_INTRMASK_DSP_AI | OS_INTRMASK_DSP_ARAM | OS_INTRMASK_DSP_DSP)
+
+#define OS_INTRMASK_EXI_0                                                      \
+    (OS_INTRMASK_EXI_0_EXI | OS_INTRMASK_EXI_0_TC | OS_INTRMASK_EXI_0_EXT)
+#define OS_INTRMASK_EXI_1                                                      \
+    (OS_INTRMASK_EXI_1_EXI | OS_INTRMASK_EXI_1_TC | OS_INTRMASK_EXI_1_EXT)
+#define OS_INTRMASK_EXI_2 (OS_INTRMASK_EXI_2_EXI | OS_INTRMASK_EXI_2_TC)
+#define OS_INTRMASK_EXI                                                        \
+    (OS_INTRMASK_EXI_0_EXI | OS_INTRMASK_EXI_0_TC | OS_INTRMASK_EXI_0_EXT |    \
+     OS_INTRMASK_EXI_1_EXI | OS_INTRMASK_EXI_1_TC | OS_INTRMASK_EXI_1_EXT |    \
+     OS_INTRMASK_EXI_2_EXI | OS_INTRMASK_EXI_2_TC)
+
+#define OS_INTRMASK_PI                                                         \
+    (OS_INTRMASK_PI_CP | OS_INTRMASK_PI_SI | OS_INTRMASK_PI_DI |               \
+     OS_INTRMASK_PI_RSW | OS_INTRMASK_PI_ERROR | OS_INTRMASK_PI_VI |           \
+     OS_INTRMASK_PI_PE_TOKEN | OS_INTRMASK_PI_PE_FINISH |                      \
+     OS_INTRMASK_PI_DEBUG | OS_INTRMASK_PI_HSP)
+
+#define OS_INTRMASK_PI_PE (OS_INTRMASK_PI_PE_TOKEN | OS_INTRMASK_PI_PE_FINISH)
 
 typedef void (*OSInterruptHandler)(__OSInterrupt, OSContext*);
 
-extern u32 __OSLastInterruptSrr0;
-extern s16 __OSLastInterrupt;
-extern s64 __OSLastInterruptTime;
+extern volatile u32 __OSLastInterruptSrr0;
+extern volatile s16 __OSLastInterrupt;
+extern volatile s64 __OSLastInterruptTime;
```

Comment:
Also not immediately important but how do you want to handle `volatile` across different platforms?

Reply:
I don't think we'll need to, the semantics of volatile are pretty well defined.